packages/angular_devkit/build_angular/src/tools/esbuild/angular/compiler-plugin.ts

```diff
@@ -61,8 +61,11 @@ export function createCompilerPlugin(
       if (!preserveSymlinks) {
         // Use the real path of the tsconfig if not preserving symlinks.
         // This ensures the TS source file paths are based on the real path of the configuration.
+        // NOTE: promises.realpath should not be used here since it uses realpath.native which
+        // can cause case conversion and other undesirable behavior on Windows systems.
+        // ref: https://github.com/nodejs/node/issues/7726
         try {
-          tsconfigPath = await realpath(tsconfigPath);
+          tsconfigPath = realpathSync(tsconfigPath);
```

Comment:
NIT: This should no longer be needed now, since tsconfig path is already resolved using the workspace realpath, which is part of the options normalisation.

Reply:
Left it there just to be safe.
I'll also be doing a round of path normalization cleanup where we can address this as well.

Reply:
Tested again, and this is indeed no longer required after https://github.com/angular/angular-cli/pull/26453/commits/23b89fb1de5edda16055c5fc65d2d9256ca0d27d